diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0e9968b..9cf9eed 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,7 +32,7 @@ set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads)
 
 find_package(FFmpeg
-    COMPONENTS AVCODEC AVFORMAT AVUTIL AVDEVICE AVFILTER SWSCALE SWRESAMPLE REQUIRED)
+    COMPONENTS AVCODEC AVFORMAT AVUTIL AVFILTER SWSCALE SWRESAMPLE REQUIRED)
 
 add_subdirectory(src)
 
diff --git a/cmake/FindFFmpeg.cmake b/cmake/FindFFmpeg.cmake
index 2746b39..cd333a2 100644
--- a/cmake/FindFFmpeg.cmake
+++ b/cmake/FindFFmpeg.cmake
@@ -166,7 +166,6 @@ if (NOT FFMPEG_LIBRARIES)
   # Check for all possible component.
   find_component(AVCODEC    libavcodec    avcodec  libavcodec/avcodec.h)
   find_component(AVFORMAT   libavformat   avformat libavformat/avformat.h)
-  find_component(AVDEVICE   libavdevice   avdevice libavdevice/avdevice.h)
   find_component(AVUTIL     libavutil     avutil   libavutil/avutil.h)
   find_component(AVFILTER   libavfilter   avfilter libavfilter/avfilter.h)
   find_component(SWSCALE    libswscale    swscale  libswscale/swscale.h)
@@ -229,7 +228,7 @@ if (NOT TARGET FFmpeg::FFmpeg)
 endif()
 
 # Now set the noncached _FOUND vars for the components.
-foreach (_component AVCODEC AVDEVICE AVFORMAT AVUTIL POSTPROCESS SWSCALE)
+foreach (_component AVCODEC AVFORMAT AVUTIL POSTPROCESS SWSCALE)
   set_component_found(${_component})
 endforeach ()
 
diff --git a/src/avutils.cpp b/src/avutils.cpp
index 6b7aa96..979a003 100644
--- a/src/avutils.cpp
+++ b/src/avutils.cpp
@@ -177,7 +177,6 @@ void init()
 #if LIBAVFILTER_VERSION_MAJOR < 7 // FFmpeg 4.0
     avfilter_register_all();
 #endif
-    avdevice_register_all();
 
 #if LIBAVCODEC_VERSION_MAJOR < 58 // FFmpeg 4.0
     av_lockmgr_register(avcpp_lockmgr_cb);
diff --git a/src/ffmpeg.h b/src/ffmpeg.h
index a01631e..f7cad6a 100644
--- a/src/ffmpeg.h
+++ b/src/ffmpeg.h
@@ -9,7 +9,7 @@ extern "C"
 #include <libavutil/mathematics.h>
 #include <libavutil/opt.h>
 #include <libavutil/pixdesc.h>
-#include <libavdevice/avdevice.h>
+#include <libavformat/avformat.h>
 #include <libswscale/swscale.h>
 #include <libswresample/swresample.h>
 #include <libavformat/version.h>
diff --git a/src/libavcpp.pc.in b/src/libavcpp.pc.in
index e3a5e00..3aaa52b 100644
--- a/src/libavcpp.pc.in
+++ b/src/libavcpp.pc.in
@@ -6,7 +6,7 @@ includedir=${prefix}/@INC_INSTALL_DIR@
 Name: libavcpp
 Description: FFmpeg libraries C++ interface
 Version: @AVCPP_VERSION@
-Requires: libavcodec >= @AVCODEC_VERSION@, libavformat >= @AVFORMAT_VERSION@, libavutil >= @AVUTIL_VERSION@, libavdevice >= @AVDEVICE_VERSION@, libavfilter >= @AVFILTER_VERSION@, libswscale >= @SWSCALE_VERSION@, libswresample >= @SWRESAMPLE_VERSION@
+Requires: libavcodec >= @AVCODEC_VERSION@, libavformat >= @AVFORMAT_VERSION@, libavutil >= @AVUTIL_VERSION@, libavfilter >= @AVFILTER_VERSION@, libswscale >= @SWSCALE_VERSION@, libswresample >= @SWRESAMPLE_VERSION@
 Requires.private:
 Conflicts:
 Libs: -L${libdir}  -lavcpp
